tests/test_docker-compose.sh

```diff
@@ -0,0 +1,660 @@
+    local rc=0
+
+    prepare_config
+    prepare_expected_file_tree
+    prepare_docker_mock
+
+    # For this test, we need 'docker images...' to return something consistent
+    # between runs that we can easily refer to.
+    cat << EOF > "${WORKDIR}/bin/docker"
+case "\$1" in
+     --version|version)
+        exit 0
+        ;;
+     compose)
+        exit 1
+        ;;
+     images)
+        # return the queried image reference as its ID
+        echo "\$4"
+        ;;
+esac
+echo "\$(basename \$0) \$@" >> "$CMDLINE_LOGGER_LOG_FILE"
+exit 0
+EOF
+
+    "${SRCDIR}/docker-compose" ArtifactInstall "${WORKDIR}/artifact" > "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "First ArtifactInstall failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+    "${SRCDIR}/docker-compose" ArtifactCommit "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "First ArtifactCommit failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    "${SRCDIR}/docker-compose" Cleanup "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "First Cleanup failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    # Now we need to change the file tree to be a new artifact
+    rm -rf "${WORKDIR}/artifact/tmp/"*
+    cat << EOF > "${WORKDIR}/artifact/header/header-info"
+{
+  "artifact_provides": { "artifact_name": "test-artifact2" }
+}
+EOF
+
+    rm "${WORKDIR}/images/image2.tar"
+    tar -C "${WORKDIR}" -czf "${WORKDIR}/artifact/files/images.tar.gz" images
+    cat << EOF > "${WORKDIR}/manifests/docker-compose.yml"
+services:
+  lighttpd:
+    image: some/lighttpd:latest
+    ports:
+      - "8081:80"
+EOF
+    tar -C "${WORKDIR}" -cf "${WORKDIR}/artifact/files/manifests.tar" manifests
+
+    "${SRCDIR}/docker-compose" ArtifactInstall "${WORKDIR}/artifact" > "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Second ArtifactInstall failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+    "${SRCDIR}/docker-compose" ArtifactCommit "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Second ArtifactCommit failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    "${SRCDIR}/docker-compose" Cleanup "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Second Cleanup failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    cat << EOF | diff -u - "$CMDLINE_LOGGER_LOG_FILE" || rc=$?
+docker image load
+docker image load
```

Comment:
By the way, these docker load _nothing_  I've been seeing and are hard to verify that are called correctly. In this test particularly as we load different images it would have value.

With [the suggestion for no stdin](https://github.com/mendersoftware/mender-container-modules/pull/2#discussion_r2410414102) this could be verified.

Reply:
Yup, I cannot disagree. I'll change it in a follow-up commit. (just to avoid complicating rebasing for me)